src/rules/unit-no-unknown/__test__/index.js

```diff
@@ -0,0 +1,256 @@
+  }, {
+    code: "a { color: rgb(255pix, 0, 51); }",
+    message: messages.rejected("pix"),
+    line: 1,
+    column: 16,
+  }, {
+    code: "a { color: hsl(255pix, 0, 51); }",
+    message: messages.rejected("pix"),
+    line: 1,
+    column: 16,
+  }, {
+    code: "a { color: rgba(255pix, 0, 51, 1); }",
+    message: messages.rejected("pix"),
+    line: 1,
+    column: 17,
+  }, {
+    code: "a { color: hsla(255pix, 0, 51, 1); }",
+    message: messages.rejected("pix"),
+    line: 1,
+    column: 17,
+  }, {
+    code: "a { margin: calc(13pix + 10px); }",
+    message: messages.rejected("pix"),
+    line: 1,
+    column: 18,
+  }, {
+    code: "a { -webkit-transition-delay: 10pix; }",
+    message: messages.rejected("pix"),
+    line: 1,
+    column: 31,
+  }, {
+    code: "a { margin: -webkit-calc(13pix + 10px); }",
+    message: messages.rejected("pix"),
+    line: 1,
+    column: 26,
+  }, {
+    code: "a { margin: some-function(13pix + 10px); }",
+    message: messages.rejected("pix"),
+    line: 1,
+    column: 27,
+  }, {
+    code: "root { --margin: 10pix; }",
+    message: messages.rejected("pix"),
+    line: 1,
+    column: 18,
+  }, {
+    code: "root { --margin: 10px + 10pix; }",
+    message: messages.rejected("pix"),
+    line: 1,
+    column: 25,
+  } ],
+})
+
+testRule(rule, {
+  ruleName,
+  syntax: "scss",
+  config: [undefined],
+
+  accept: [{
+    code: "a { width: 1em; \n// width: 10pix\n }",
+    description: "ignore wrong unit within comments",
+  }],
+
+  reject: [ {
+    code: "a { margin: 10pix + 10px; }",
+    message: messages.rejected("pix"),
+    line: 1,
+    column: 13,
+  }, {
+    code: "a { $margin: 10pix; }",
+    message: messages.rejected("pix"),
+    line: 1,
+    column: 14,
+  }, {
+    code: "a { $margin: 10px + 10pix; }",
+    message: messages.rejected("pix"),
+    line: 1,
+    column: 21,
+  }, {
+    code: "a { margin: $margin + 10pix; }",
+    message: messages.rejected("pix"),
+    line: 1,
+    column: 23,
+  }, {
+    code: "$breakpoints: ( small: 767px, medium: 992pix, large: 1200px );",
+    message: messages.rejected("pix"),
+    line: 1,
+    column: 39,
+  }, {
+    code: "a { font: (italic bold 10px/8pix) }",
+    message: messages.rejected("pix"),
+    line: 1,
+    column: 29,
+  }, {
+    code: "font: 14pix/#{$line-height};",
+    message: messages.rejected("pix"),
+    line: 1,
+    column: 7,
+  } ],
+})
+
+testRule(rule, {
+  ruleName,
+  syntax: "less",
+  config: [undefined],
+
+  accept: [{
+    code: "a { width: 1em; \n// width: 10pix\n }",
+    description: "ignore wrong unit within comments",
+  }],
+
+  reject: [{
+    code: "@variable: 10pix",
+    message: messages.rejected("pix"),
+    line: 1,
+    column: 12,
+  }],
+})
+
+testRule(rule, {
+  ruleName,
+  config: [ undefined, { ignore: ["pix"] } ],
+
+  accept: [ {
+    code: "a { margin: 10px; }",
+  }, {
+    code: "a { margin: 10pix; }",
+  } ],
+
+  reject: [{
+    code: "a { margin: 10pixels; }",
+  }],
+})
```

Comment:
@davidtheclark I know about it, I think work it out after the rule accepted. Make this in all rules unit-*